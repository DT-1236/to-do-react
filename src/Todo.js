import React, { Component } from 'react';

class Todo extends Component {
  // add remove function from the parent TodoList
  constructor(props) {
    super(props);
    this.handleRemove = this.handleRemove.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      isEdit: false,
      content: this.props.task
    };
  }
  handleRemove() {
    this.props.remove(this.props.id);
  }

  //handleEdit
  handleEdit() {
    //this.props.edit(this.props.id);
    // console.log(this.props.editTask);
    if (this.state.isEdit) {
      this.props.editTask(this.props.id, this.state.content);
      this.setState({ isEdit: false });
    } else {
      this.setState({ isEdit: true });
    }
  }
  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  render() {
    const display = this.state.isEdit ? (
      <input
        onChange={this.handleChange}
        name="content"
        value={this.state.content}
        type="text"
      />
    ) : (
      this.props.task
    );
    return (
      <li>
        {display}
        <button name="delete" onClick={this.handleRemove}>
          delete
        </button>
        <button name="edit" onClick={this.handleEdit}>
          {this.state.isEdit ? 'Done' : 'Edit'}
        </button>
      </li>
    );
  }
}
export default Todo;
