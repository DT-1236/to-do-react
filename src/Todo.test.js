import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Todo from './Todo';
//in src/**-test.js

//smokeTest

describe('simple smoking snapshots', function() {
  it('renders without crushing', () => {
    shallow(<Todo />);
  });

  it('matches snapshot', function() {
    let wrapper = shallow(<Todo />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('the inital sstate is set correctly', () => {
  const wrapper = shallow(<Todo />);

  expect(wrapper.state().isEdit).toBe(false);
});

describe('the todo has two buttons and', () => {
  it('the todo has two button', () => {
    const wrapper = shallow(<Todo />);
    let buttons = wrapper.find('button');
    expect(buttons).toHaveLength(2);
  });

  it('edit button changes the state', () => {
    const wrapper = shallow(<Todo />);
    let editButton = wrapper.find("button[name='edit']");
    editButton.simulate('click');
    expect(wrapper.state().isEdit).toBe(true);
  });

  it('edit mode, change the input', () => {
    const wrapper = shallow(<Todo />);
    let editButton = wrapper.find("button[name='edit']");
    editButton.simulate('click');
    expect(wrapper.state().isEdit).toBe(true);
    let input = wrapper.find('input');
    input.simulate('change', { target: { name: 'content', value: 'a' } });
    input = wrapper.find('input');
    expect(input.props().value).toBe('a');
    expect(wrapper.state().content).toBe('a');
  });
});
