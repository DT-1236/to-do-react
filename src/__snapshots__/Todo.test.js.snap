// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple smoking snapshots matches snapshot 1`] = `
<li>
  <button
    name="delete"
    onClick={[Function]}
  >
    delete
  </button>
  <button
    name="edit"
    onClick={[Function]}
  >
    Edit
  </button>
</li>
`;
